frame/asset-conversion/src/lib.rs

```diff
@@ -352,24 +352,24 @@ pub mod pallet {
 	#[pallet::hooks]
 	impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {
 		fn integrity_test() {
-			sp_std::if_std! {
-				sp_io::TestExternalities::new_empty().execute_with(|| {
-					// ensure that the `AccountId` is set properly and doesn't generate the same
-					// pool account for different pool ids.
-					let native = T::MultiAssetIdConverter::get_native();
-					// Decode the asset ids from bytes.
-					let asset_1 = T::MultiAssetIdConverter::into_multiasset_id(&T::AssetId::decode(&mut vec![0u8, 0, 0, 1].as_slice()).unwrap());
-					let asset_2 = T::MultiAssetIdConverter::into_multiasset_id(&T::AssetId::decode(&mut vec![255u8, 255, 255, 255].as_slice()).unwrap());
-					assert!(asset_1 != asset_2, "unfortunatly decoded to be the same asset.");
-					let pool_account_1 = Self::get_pool_account(&(native.clone(), asset_1));
-					let pool_account_2 = Self::get_pool_account(&(native, asset_2));
-					assert!(sp_std::mem::size_of::<T::AccountId>() >= sp_std::mem::size_of::<u128>());
-					assert!(
-						pool_account_1 != pool_account_2,
-						"AccountId should be set at least to u128"
-					);
-				});
-			}
+			#[cfg(test)]
```

Comment:
I think this line is superfluous.

Reply:
It won't compile without it (`associated type `BenchmarkHelper` not found`). I think the `integrity_test` is not just called as a test? For the purposes here it's sufficient if this check fails at test time.

Reply:
Fixed by https://github.com/paritytech/substrate/pull/14298